client/app/components/dashboards/CreateDashboardDialog.jsx

```diff
@@ -36,7 +37,7 @@ function CreateDashboardDialog({ dialog }) {
     if (name !== "") {
       setSaveInProgress(true);
 
-      $http.post("api/dashboards", { name }).then(({ data }) => {
+      axios.post("api/dashboards", { name }).then(({ data }) => {
```

Comment:
I think that things like this should be moved to services (e.g. have a method called `updateDashboard` of whatever) - there should be no direct HTTP calls in components. There are a lot of similar places in other components as well

Reply:
You're right. It was wrong to use `$http` all over. But wherever things are just working, we should keep things as they were (except for adopting `axios` so we can drop `$http`). In parallel, let's decide on a pattern we want to use for our API calls/resources, that will be more inline with how React works (returning immutable objects for example). Any suggestions for such pattern? Let's discuss in the issue.

Wherever things are not working as is or we have to jump through hoops to make it work, we can start adopting the new pattern. The rest we will update when we are done with the React migration.